test/lib.sh

```diff
@@ -59,3 +59,98 @@ EOF
 hr() {
     echo "========================="
 }
+
+## diff BASELINE
+#
+# Recursively compare the current repository content to the BASELINE repository
+#
+# :param BASELINE: An existing directory used as reference to compared sync'ed files
+diff() {
+	for file in `ls` ; do 
+		if [ -d $file ] ; then 
+			pushd $file  > /dev/null
+			diff $1/$file 
+			popd  > /dev/null
+		elif [ -f $file ] ; then
+			cmp $file $1/$file
+			if [ ! $? = 0 ] ; then
+				handle_error_counter INC
+			fi
+		fi
+	done
+}
+
+## compare FOLDER
+#
+# Check FOLDER is properly sync'ed, determining the reference base on FOLDER 
+#
+# :param FOLDER: An existing directory sync'ed by boilerplate and to be checked
+compare() {
+	if [ $1 = "_data" ] ; then
+		if [ ! -f $repo/boilerplate/_data/last_boilerplate_commit ] ; then
+			# TODO: Check the content of the file to ensure it contains the proper commit in addition to the file existence
+			handle_error_counter INC
+		fi
+	else
+		pushd $1  > /dev/null
+		diff $BOILERPLATE_GIT_REPO/boilerplate/$1
+		popd > /dev/null
+	fi
+}
+
+## check_update
+#
+# Check the boilerplate synchronization is properly working, covering generics and convention
+# specific parts
+check_update() {
+	handle_error_counter INIT
+	pushd $repo/boilerplate > /dev/null
+	
+	compare _data
+	compare _lib
+	
+	while read convention ; do
+	  if [ -d $BOILERPLATE_GIT_REPO/boilerplate/$convention ] ; then
+		  compare $convention
+	  fi
+	done < $repo/boilerplate/update.cfg
+	
+	popd > /dev/null
+	
+	handle_error_counter CHECK
+	
+	exit $?
```

Comment:
This should be
```suggestion
	return $?
```
Let the caller decide whether it's time to `exit`.